TvarkauVilniu/App/Scenes/Main/ViewController.swift

```diff
@@ -80,4 +80,34 @@ extension ViewController: UIWebViewDelegate {
         return true
     }
 
+    func webViewDidStartLoad(_ webView: UIWebView) {
+        loadCookies()
+    }
+
+    func webViewDidFinishLoad(_ webView: UIWebView) {
+        saveCookies()
+    }
+    
+    func saveCookies() {
+        guard let cookies = HTTPCookieStorage.shared.cookies else {
+            return
+        }
+        let array = cookies.compactMap { (cookie) -> [HTTPCookiePropertyKey: Any]? in
+            cookie.properties
+        }
+        UserDefaults.standard.set(array, forKey: "cookies")
+        UserDefaults.standard.synchronize()
+    }
+
+    func loadCookies() {
+        guard let cookies = UserDefaults.standard.value(forKey: "cookies") as? [[HTTPCookiePropertyKey: Any]] else {
+            return
+        }
+        cookies.forEach { (cookie) in
+            guard let cookie = HTTPCookie.init(properties: cookie) else {
```

Comment:
forEach and guard could be combined to compactMap on cookies array and then iterated over with ‘forEach(HTTPCookieStorage.shared.setCookie)’